### PR TITLE
Add geolocation changes

### PR DESCRIPTION
**Pull request by hivepress**

### CODE REVIEW

File: assets/js/common.js [hunk 1]
```diff
@@ -264,6 +264,13 @@
 							},
 						};
 
+					if (container.data('icon')) {
```
Comment (hivepress) [inline assets/js/common.js:267]: Please follow the task description if it describes near-exact implementation, we need support for data-options instead of targeting a single data-icon attribute, then we'll be able to add future changes without extra code. For example, sometimes people are asking how to change some defaults and while it's possible via PHP action and filters hook, it's not possible in JS, so adding support for passing a settings JSON array via data-options is needed

File: assets/js/common.js [hunk 2]
```diff
@@ -262,7 +262,17 @@
 								lat: data.latitude,
 								lng: data.longitude,
 							},
-						};
+						},
+						options = container.data('options');
```
Comment (kuznev) [inline assets/js/common.js:266]: это перенеси выше условий на проверку какой провайдер карт выбран, так как эти опции могут использоваться не только для гугл карт

File: assets/js/common.js [hunk 3]
```diff
@@ -264,6 +265,13 @@
 							},
 						};
 
+					if (options && options.icon) {
```
Comment (hivepress) [inline assets/js/common.js:268]: It's still doesn't match the initial idea (e.g. it's not possible to override other settings via data-options, only the icon) but ok -- we can add a specific HTML attribute for this custom work, please change it to a specific attribute data-marker which passes the marker icon URL.

Comment (hivepress) [reply assets/js/common.js:268]: Also please remember to test if map clustering (merging into a group when it's zoomed out) still works with a custom marker.

File: assets/js/common.js [hunk 4]
```diff
@@ -264,6 +265,38 @@
...
+							overlay.draw = function() {
+								var position = this.getProjection().fromLatLngToDivPixel(marker.getPosition());
+
+								layer.style.left = position.x + 'px';
+								layer.style.top = position.y + 'px';
+							};
+						};
+
+						overlay.setMap(map);
+					}
```
Comment (kuznev) [inline assets/js/common.js:291]: Please revert these changes and show price in the marker on click as it was before
